### PR TITLE
Add rule to build Nextflow docker image for ARM

### LINKED ISSUE #2750: Add docker builds on non-ARM architectures
## New feature

Hi, 

Recent builds (22.x and higher) of the [official nextflow image](https://hub.docker.com/r/nextflow/nextflow/tags) are only built for the arm64 architecture. Would it be possible to additionally provide builds for other architectures, particularly amd64/x86_64? 

Thanks!

## Usage scenario 

Running nextflow on amd64-based machines, such as AWS EC2 instances and non-M1 macs

## Suggest implementation 

Use [docker buildx](https://docs.docker.com/engine/reference/commandline/buildx_build/), as suggested in [official docs](https://www.docker.com/blog/multi-platform-docker-builds/), e.g.

```make
build:
	docker buildx build \
		--platform linux/amd64,linux/arm64,linux/arm/v7 \
		--tag nextflow/nextflow:${version} .
```

### PR DESCRIPTION
**Pull request by bentsherman**
Close #2750 

Haven't tested it myself, just drafting a PR for anyone who wants to try a Nextflow docker image on ARM:
```bash
cd docker
make build-arm
```

### CODE REVIEW

File: docker/Makefile [hunk 1]
```diff
@@ -15,10 +15,14 @@
...
 version ?= none
 
-build: dist/docker
+build: dist/docker/amd64
 	cp ../nextflow .
 	docker buildx build --platform linux/amd64 --output=type=docker --progress=plain --tag nextflow/nextflow:${version} .
 
+build-arm: dist/docker/arm64
+	cp ../nextflow .
+	docker buildx build --platform linux/arm64 --output=type=docker --progress=plain --tag nextflow/nextflow:${version} .
```
Comment (mribeirodantas) [inline docker/Makefile:24]:
```suggestion
	docker buildx build --platform linux/arm64 --output=type=docker --progress=plain --tag nextflow/nextflow:${version} --build-arg TARGETPLATFORM=linux/arm64 .
```

**Review by mribeirodantas — CHANGES_REQUESTED** [review]
The Dockerfile remains with the target platform set [by default] to amd64. With this change, I was able to create the container image in my machine (Apple M1).

File: docker/Makefile (hunk 1 above)
Comment (mribeirodantas) [reply docker/Makefile:24]: Yeah! 🥳

CleanShot 2023-06-14 at 17 27 09@2x

No warning about architecture 🤩

Comment (mribeirodantas) [reply docker/Makefile:24]: Sure.

CleanShot 2023-06-15 at 02 37 45@2x

CleanShot 2023-06-15 at 02 37 03@2x

**Review by mribeirodantas — APPROVED** [review]

File: docker/Makefile [hunk 2]
```diff
@@ -15,10 +15,14 @@
 
 version ?= none
 
-build: dist/docker
+build: dist/docker/amd64
 	cp ../nextflow .
 	docker buildx build --platform linux/amd64 --output=type=docker --progress=plain --tag nextflow/nextflow:${version} .
```
[the remark this answers is not in the document]
Comment (bentsherman) [reply docker/Makefile:20]: Sure, that is the default in the Dockerfile but the consistency is nice

**Review by pditommaso — APPROVED** [review]

**Pull request merged**